broken-site/broken-site-impl/src/main/java/com/duckduckgo/brokensite/impl/BrokenSitePomptDataStore.kt

```diff
@@ -138,6 +154,57 @@ class SharedPreferencesDuckPlayerDataStore @Inject constructor(
         return dismissStreak.first()
     }
 
+    override suspend fun addDismissal(dismissal: LocalDateTime) {
+        store.edit { prefs ->
+            prefs[DISMISS_EVENTS] = (prefs[DISMISS_EVENTS]?.toSet() ?: emptySet()).plus(formatter.format(dismissal))
+        }
+    }
+
+    override suspend fun getDismissalCountBetween(
+        t1: LocalDateTime,
+        t2: LocalDateTime,
+    ): Int {
+        val allDismissEvents = store.data.map { prefs ->
+            prefs[DISMISS_EVENTS]?.toSet() ?: emptySet()
+        }.first()
+
+        return allDismissEvents.count { dateString: String ->
+            try {
+                val eventDateTime = LocalDateTime.parse(dateString, formatter)
+                eventDateTime.isAfter(t1) && eventDateTime.isBefore(t2)
+            } catch (e: Exception) {
+                false
+            }
+        }
+    }
+
+    override suspend fun deleteAllExpiredDismissals(expiryDate: String) {
+        val normalizedExpiry = LocalDateTime.ofInstant(Instant.parse(expiryDate), ZoneOffset.UTC)
```

Comment:
I think it's best to have zoneID passed as a parameter as well, for clarity and extra safety